### PR TITLE
Now enable bayesian LS on pom.xml (after the rebase on master)

### PR DESCRIPTION
**Pull request by davidfestal**
This is related to redhat-developer/rh-che#218

With this change, I tested (on OSIO) that the Bayesian plugin now displays recommendations on `pom.xml` files as it does on `package.json` files. 

Signed-off-by: David Festal <dfestal@redhat.com>

### CODE REVIEW

File: builds/fabric8-che/plugins/che-plugin-bayesian-lang-server/src/main/java/org/eclipse/che/plugin/languageserver/bayesian/server/launcher/BayesianLanguageServerLauncher.java [hunk 1]
```diff
@@ -87,8 +87,10 @@ private static LanguageServerDescription createServerDescription() {
                                                                               null),//
                                                            new DocumentFilter(JsonModule.LANGUAGE_ID,//
                                                                               "package\\.json",//
-                                                                              null)//
-                                             // TODO pom.xml
+                                                                              null),
+                                                           new DocumentFilter("pom",
```
Comment (sunix) [inline builds/fabric8-che/plugins/che-plugin-bayesian-lang-server/src/main/java/org/eclipse/che/plugin/languageserver/bayesian/server/launcher/BayesianLanguageServerLauncher.java]: Isn't it a constant you could reuse ?

Comment (davidfestal) [inline builds/fabric8-che/plugins/che-plugin-bayesian-lang-server/src/main/java/org/eclipse/che/plugin/languageserver/bayesian/server/launcher/BayesianLanguageServerLauncher.java]: If such a constant had been available in the maven language server I wold have taken it, but unfortunately the related code in the maven language server launcher is the following:

```java
    @Override
    public LanguageServerDescription getDescription() {
        return new LanguageServerDescription("org.eclipse.che.plugin.maven", Collections.singletonList("pom"), Collections.emptyList());
    }
```

On the other hand `pom` is quite and expected string for a pom file ;-)

Comment (tsmaeder) [inline builds/fabric8-che/plugins/che-plugin-bayesian-lang-server/src/main/java/org/eclipse/che/plugin/languageserver/bayesian/server/launcher/BayesianLanguageServerLauncher.java]: It probably makes sense to extract the launguage id "pom" to a constant when we start using it outside of the maven module.

Comment (davidfestal) [inline builds/fabric8-che/plugins/che-plugin-bayesian-lang-server/src/main/java/org/eclipse/che/plugin/languageserver/bayesian/server/launcher/BayesianLanguageServerLauncher.java]: Well, in this case we'll have to wait for the upstream PR to be merged first.
I'd prefer doing this in a second PR synchronized with a related upstream Che PR.
Is it OK for you ?

Comment (sunix) [inline builds/fabric8-che/plugins/che-plugin-bayesian-lang-server/src/main/java/org/eclipse/che/plugin/languageserver/bayesian/server/launcher/BayesianLanguageServerLauncher.java]: yes do 2 separates PR (che and rh-che) for that thanks

**Pull request merged**